crates/ruff_python_formatter/src/expression/expr_attribute.rs

```diff
@@ -169,16 +163,25 @@ impl NeedsParentheses for ExprAttribute {
             == CallChainLayout::Fluent
         {
             OptionalParentheses::Multiline
-        } else if context
-            .comments()
-            .dangling(self)
-            .iter()
-            .any(|comment| comment.line_position().is_own_line())
-            || context.comments().has_trailing_own_line(self)
-        {
+        } else if context.comments().has_dangling(self) {
             OptionalParentheses::Always
         } else {
             self.value.needs_parentheses(self.into(), context)
         }
     }
 }
+
+fn find_only_token_in_range(
+    range: TextRange,
+    token_kind: SimpleTokenKind,
+    source: &str,
+) -> SimpleToken {
+    let mut tokens = SimpleTokenizer::new(source, range)
+        .skip_trivia()
+        .skip_while(|token| token.kind == SimpleTokenKind::RParen);
+    let token = tokens.next().expect("Expected a token");
+    debug_assert_eq!(token.kind(), token_kind);
+    let mut tokens = tokens.skip_while(|token| token.kind == SimpleTokenKind::LParen);
+    debug_assert_eq!(tokens.next(), None);
+    token
+}
```

Comment:
This is the same as the method in placement, but it doesn't have a natural home. Any ideas?

Reply:
you could make it a `SimpleTokenizer` method